internal/clients/imagebuilder/image_builder_client.go

```diff
@@ -0,0 +1,80 @@
+package imagebuilder
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+)
+
+const URL = "https://console.redhat.com/api/image-builder/v1/compose"
+
+func RequestImageBuilder(client *http.Client, endpoint string, method string, orgId string, body *strings.Reader) ([]byte, error) {
+	var req *http.Request
+	var err error
+
+	req, err = http.NewRequest(method, endpoint, body)
+	if err != nil {
+
```

Comment:
@lzap So I saw you used regular `fmt.ErrorOf in /clouds.`
how do you want to deal with logging errors? 
(This function will be called as part of the sources service)
 The same question goes for sources_client.go

Reply:
So in my current proposal I have a type named ec2.Client which encapsulates AWS SDK client connection, Context as well as Zerolog logger from the context. Let's definitely use the logger from context which also has context information (request id etc). For now, I am using fmt.Errorf for simplicity, but going forward I think we should declare our own error types because clients will eventually do error out at some point and we need to handle them nicely.

https://github.com/RHEnVision/provisioning-backend/pull/16/files#diff-a2ed92d8fa157beb7e8104db92ae53957c91a288f949aea765823bdb2025a651